R/helpers.R

```diff
@@ -10,41 +10,36 @@
 #' @param valuecol, name of the column containing the GDP values, default is "weight".
 #' @param usecache, store the result in a RDS file in the working directory, default is FALSE.
 #' @param gdpfile, if caching is required, specify the filename here, default is "GDPcache.rds"
+#' @param isolev, if the data provided has to be on ISO or regional level
 #' @keywords gdp
 #' @import data.table
 #' @importFrom madrat calcOutput
 #' @export
 
-getRMNDGDP <- function(scenario="gdp_SSP2",
-                      yearcol="year",
-                      isocol="iso",
-                      valuecol="weight",
-                      usecache=F,
-                      gdpfile="GDPcache.rds"){
-
-    variable <- Year <- NULL
-
-    if(usecache && file.exists(gdpfile)){
-        cat("getGDP_dt: Using cached GDP data in", gdpfile, "\n")
-        return(readRDS(gdpfile))
-    }
-
-    GDPppp_country <- calcOutput("GDPppp", aggregate = F)[,, scenario]
-
-    gdp <- as.data.table(GDPppp_country)[variable == scenario]
-    gdp[, (yearcol) := as.numeric(gsub("y", "", Year))][, Year := NULL]
-    setnames(gdp, c("ISO3", "value"), c(isocol, valuecol))
+getRMNDGDPcap <- function(scenario="gdp_SSP2",
+                          yearcol="year",
+                          isocol = "region",
+                          isolev = F,
+                          valuecol="weight",
+                          usecache=F,
+                          gdpfile="GDPcache.rds"){
 
-    if(usecache){
-        saveRDS(gdp, gdpfile)
-    }
+  `.` <- iso <- value <- GDP_cap <- weight <- POP_val <- region <- NULL
```

Comment:
if the cache is not required anymore, please remove the respective arguments from the function (usecache, gdpfile)

Reply:
so is this function not called iteratively?